packages/plugins/plugin-firebase/src/methods/identify.ts

```diff
@@ -6,6 +6,27 @@ export default async (event: IdentifyEventType) => {
     await firebaseAnalytics().setUserId(event.userId!);
   }
   if (event.traits) {
-    await firebaseAnalytics().setUserProperties(event.traits as any);
+    let eventTraits = event.traits;
+    let safeTraits: Record<string, string>;
+
+    const checkType = (value: any) => {
+      return typeof value === 'object' && !Array.isArray(value);
+    };
+    safeTraits = Object.keys(eventTraits).reduce(
+      (acc: Record<string, string>, trait) => {
+        if (checkType(eventTraits[trait])) {
+          console.log(
+            'We detected an object or array data type. Firebase does not accept nested traits.'
+          );
+        }
+        if (trait !== undefined) {
+          acc[trait] = eventTraits[trait]!.toString();
```

Comment:
for safety it might be good not to force unwrap `eventTraits[trait]`. Technically it wouldn't ever happen since you are iterating over keys but it's a good practice for maintainability:
```
if (trait !== undefined && eventTraits[trait] !== undefined) {
  ...
```

Reply:
there was a type error here until so i recommended force unwrapping since we can assume it will always be there (and said it's not good practice in most cases)